org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -156,27 +159,41 @@ public boolean isBuildFile(IResource resource) {
 		return buildSupports().filter(bs -> bs.isBuildFile(resource)).findAny().isPresent();
 	}
 
-	private IProjectImporter getImporter(File rootFolder, IProgressMonitor monitor) throws InterruptedException, CoreException {
+	private IProjectImporter getImporter(File rootFolder, IProgressMonitor monitor) throws OperationCanceledException, CoreException {
 		Collection<IProjectImporter> importers = importers();
 		SubMonitor subMonitor = SubMonitor.convert(monitor, importers.size());
+		IProjectImporter preferred = null;
+		int preferredRelevance = -1;
 		for (IProjectImporter importer : importers) {
 			importer.initialize(rootFolder);
-			if (importer.applies(subMonitor.split(1))) {
-				return importer;
+			int relevance = importer.applies(subMonitor.split(1));
```

Comment:
I find that approach a bit dubious. If a gradle project has a maven pom.xml, who wins?